### PR TITLE
Publish npm packages with provenance

### PR DESCRIPTION
**Pull request by koddsson**
This will allow us to have a cool badge on npm.

Example:
image

### CODE REVIEW

File: .github/workflows/npm-publish.yml [hunk 1]
```diff
@@ -23,13 +23,20 @@ jobs:
...
         with:
-          node-version: 20
-          registry-url: https://registry.npmjs.org/
+          node-version: 22.x
+          registry-url: "https://registry.npmjs.org"
+          cache: "npm"
       - run: npm ci
-      - run: npm publish
+      - run: npm run build --if-present
+      - run: npm version ${TAG_NAME} --git-tag-version=false
```
Comment (43081j) [inline .github/workflows/npm-publish.yml:37]: Does this mean we can create releases without having to push a version bump to main? 

Will we lose the tags and version in our package file by doing this? Sounds nice being able to release without the extra push but also I like having tags and the right version in the JSON 😅

Comment (keithamus) [reply .github/workflows/npm-publish.yml:37]: > Does this mean we can create releases without having to push a version bump to main?

Yes.

> Will we lose the tags and version in our package file by doing this? Sounds nice being able to release without the extra push but also I like having tags and the right version in the JSON 😅

Also yes. Typically I set package.json version to `0.0.0-development` so it's clear that the version is generated.

**Review by keithamus — DISMISSED** [review]
:+1: LGTM

File: .github/workflows/npm-publish.yml (hunk 1 above)
Comment (43081j) [reply .github/workflows/npm-publish.yml:37]: No worries I think that makes sense. @koddsson can you update the package version so that is clear?

**Review by 43081j — APPROVED** [review]

**Review by 43081j — APPROVED** [review]

**Pull request merged**